src/org/ods/component/Pipeline.groovy

```diff
@@ -278,33 +280,14 @@ class Pipeline implements Serializable {
     if (!this.bitbucketNotificationEnabled) {
       return
     }
-    if (!context.jobName || !context.tagversion || !context.credentialsId || !context.buildUrl || !context.bitbucketUrl || !context.gitCommit) {
-      logger.info "Cannot set BitBucket build status to ${state} because required data is missing!"
+    if (!context.jobName || !context.tagversion || !context.buildUrl || !context.gitCommit) {
+      logger.info "Cannot set Bitbucket build status to '${state}' because required data is missing!"
       return
     }
 
-    logger.info "Setting BitBucket build status to ${state} ..."
+    logger.info "Setting Bitbucket build status to '${state}' ..."
```

Comment:
see above (and logger.debug) .. in fact I vouch that all services get the logger injected, to finally harmonize this)

Reply:
I'm torn on this. While I have already violated my own rule a few times, I try to avoid giving logging into the service. The service just does its job (interfacing with Bitbucket), the pipeline is in control how it wants to present this to the user. E.g. "failing the build" is a similar thing which the pipeline should control. The service cannot decide whether the caller wants to stop or continue the build if the service encounters an exception.

Reply:
I moved this statement into the service now in 5650666.